content/articles/js-prototypes-vs-classes/index.md

```diff
@@ -0,0 +1,175 @@
+                }
+
+                studentDetails() {
+                    let yearOfStudy= new Date(this.yearOfStudy * 1000);
+                let months =['January','February','March','April','May','June','July','August','September','October','November','December'];
+                // get the year of study
+                let year= yearOfStudy.getFullYear();
+                //get month of study
+                let month=months[yearOfStudy.getMonth()];
+                //get day of study
+                let day = yearOfStudy.getDay();
+                //student date
+                let date= month + ' '  + day + ', ' + year;
+                //return statement
+                let output= this.studentName + ' of registration number '+ this.registrationNumber+ ' joined ' + this.schoolName + ' on '+ date + ' pursue a degree in '+
+                this.specialization;
+                return output;
+                }
+            }
+    
+        </script>
+    </head>
+    <body>
+    
+    </body>
+</html>
+```
+
+### The key differences between JavaScript classes and prototypes
+
+In the previous sections, we've looked at the JavaScript classes and prototypes, in this part, let's have a look at their differences.
+
+The first and the main point is that JavaScript is based on **prototypes** and not classes. The prototypes themselves are objects instances as have in the previous examples while classes have types that are only instantiated at runtime. This explains why JavaScript developers should only stick to prototypes that using classes.
+
+Let's look at this example:
+
+```js
+//using prototypes
+function PrototypicalStudent(name = "John Doe", school = "Example") {
+  this.name = name
+  this.school = school
+}
+
+PrototypicalStudent.prototype.details = function() {
+  return `${this.name}, ${this.school}!`
+}
+
+const studentProto = new PrototypicalStudent("John", "Doe")
+
+console.log(studentProto.details())
+```
+
+In the above class, we defined a prototypal class, let's have a look at the same implementation using classes,
+
+```js
+
+
+class ClassicalStudent {
+  constructor(name = "John Doe", school = "Example") {
+    this.name = name
+    this.school = school
+  }
+
+  studentProto() {
+    return `${this.name}, ${this.school}!`
+  }
+}
+
+const classyStudent = new ClassicalStudent("John Doe", "ExampleSchool")
+
+console.log(classyStudent.studentProto())
+
+```
+
+You quickly see a  prototypal inheritance in the first example, even though, it's more expressive in the second implementation. Nevertheless, you have seen how you can solve a problem using either case.
+
+### Conclusion
+
+In this tutorial, we've seen how classes and prototypes are being used in JavaScript. We've also seen that prototypes themselves are object instances while classes only get instantiated at runtime.  
+With this background knowledge on how both features work in JavaScript, you can now decide on when to use either of the two.  
```

Comment:
I think you've not exhaustively discussed the key differences. Kindly add them. Thanks.